src/common/components/build-row/index.js

```diff
@@ -67,33 +69,57 @@ const BuildRow = (props) => {
     ? `/user/${repository.fullName}/${buildId}`
     : null;
 
-  return (
-    <Row key={ buildId }>
-      <Data col="15">
-        { buildUrl
-          ? <Link to={buildUrl}>{`#${buildId}`}</Link>
-          : <span>{`#${buildId}`}</span>
-        }
-      </Data>
-      <Data col="20">
-        { getBuildTriggerMessage(repository, reason, commitId) }
-      </Data>
-      <Data col="15">
-        {architecture}
-      </Data>
-      <Data col="15">
-        {humanDuration}
-      </Data>
-      <Data col="35">
-        <BuildStatus
-          link={buildUrl}
-          colour={colour}
-          statusMessage={statusMessage}
-          dateStarted={dateStarted}
-        />
-      </Data>
-    </Row>
+  const buildTriggerCell = (
+    <Data col="20">
+      { getBuildTriggerMessage(repository, reason, commitId) }
+    </Data>
   );
+
+  if (isRequest) {
```

Comment:
Seems like row for build request row is quite different from build row, so maybe it would be worth creating separate `RequestRow` component for that?
This should allow to avoid having `BuildRow` returning 2 different things depending on the build/request type. Also maybe we can make `RequestRow` get proper params to avoid detecting error based on status message comparison.

Reply:
That was what I did in my first pass.  When I worked out that the internal API needs to merge builds and requests I merged them back into a single component, but maybe you're right that they'd be better separated.  I'll look into that.

The `statusMessage` comparison is IMO not as bad as it looks, because that's effectively an enumerated list controlled by BSI (see `UserFacingState`), and there's some precedent such as in `src/common/containers/build-details.js`.  I agree that it would be nicer to have this be something that's more obviously an enum rather than display text, but since it's following an existing pattern I think we could reasonably do that in a separate PR?

Reply:
Yes, I just looked into the build object and it seems we don't have anything better then `statusMessage` in there to do the comparison. So we can leave it as it is or address it later.

This also makes separating `BuildRow` and `RequestRow` less important, so I'll let you decide if that still makes sense. Thanks!

Reply:
I've separated them out now.  Thanks for the suggestion.